assets/src/util/fetchResult.ts

```diff
@@ -19,3 +19,9 @@ export const isOk = <T>(r: FetchResult<T>): r is Ok<T> => "ok" in r
 
 export const isFetchError = <T>(r: FetchResult<T>): r is FetchError =>
   "is_error" in r && r.is_error === true
+
+export const loading = <T>(): FetchResult<T> => ({ is_loading: true })
```

Comment:
```suggestion
/*
 * Constructor functions to enable a user to create FetchResult's easily
 * For instance, something like:
 *   const result = loading()
 *   const result = ok(successfulData)
 */
export const loading = <T>(): FetchResult<T> => ({ is_loading: true })
```

Nit (non-blocking): It took me a minute to figure out what these were. An explanatory comment might be helpful

Reply:
How about 22d0737a? It's in JSDoc format so it provides inline documentation in editors and stuff.

Reply:
Even better! I've never fully understood the JSDoc thing, but that's pretty neat!